### PR TITLE
chore(deps): Bump `@metamask/accounts-controller` to `^18.2.1`, `@metamask/snaps-{execution-environments,sdk,utils}` to latest

### LINKED ISSUE #11041: AccountsController upgrade from v14 to v18
### What is this about?

The `AccountsController` is behind by four major versions and needs to be updated.
Upgrade from v14 to v18.

Because this controller is pretty far behind we expect it will have a number of breaking changes that will need to be dealt with so we haven't combined this with any other controller upgrades.

### Scenario

n/a

### Design

n/a

### Technical Details

n/a

### Threat Modeling Framework

n/a

### Acceptance Criteria

- `AccountsController` is on v18

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by MajorLift**
## **Description**

```md
### Changed

- Bump `@metamask/accounts-controller` from `^18.1.0` to `^18.2.1`
- Bump `@metamask/snaps-execution-environments` from `^6.6.2` to `^6.7.2`
- Bump `@metamask/snaps-sdk` from `^6.4.0` to `^6.5.0`
- Bump `@metamask/snaps-utils` from `^8.0.1` to `^8.1.1`
```

## **Related issues**

- Closes https://github.com/MetaMask/metamask-mobile/issues/11041
- Blocked by https://github.com/MetaMask/metamask-mobile/pull/11351

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by Jonathansoufer — DISMISSED** [review]
LGTM!

**Review by Prithpal-Sooriya — DISMISSED** [review]

**Comment by owencraston** [comment]
@MajorLift I already updated a few of these packages in this [pr](https://github.com/MetaMask/metamask-mobile/pull/10829/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519).

### Already done
- "@metamask/utils": "^9.2.1", ✅ 
- "@metamask/snaps-controllers": "^9.8.0", ✅ 
- "@metamask/snaps-sdk": "^6.4.0", ✅ 

### Needs updating
- Bump `@metamask/accounts-controller` from `^18.1.0` to `^18.2.1` 🚫 
- Bump `@metamask/snaps-execution-environments` from `^6.6.2` to `^6.7.2` 🚫 
- Bump `@metamask/snaps-sdk` from `^6.4.0` to `^6.5.0` 🚫 
- Bump `@metamask/snaps-utils` from `^8.0.1` to `^8.1.1` 🚫

File: app/lib/snaps/SnapsExecutionWebView.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { createStyles } from './styles';
 import { WebViewError } from '@metamask/react-native-webview/lib/WebViewTypes';
 import { WebViewInterface } from '@metamask/snaps-controllers/react-native';
 import { PostMessageEvent } from '@metamask/post-message-stream';
+import { WebViewInterface } from '@metamask/snaps-controllers/react-native';
```
Comment (owencraston) [inline app/lib/snaps/SnapsExecutionWebView.tsx:9]: duplicate of line 7.

Comment (MajorLift) [reply app/lib/snaps/SnapsExecutionWebView.tsx:9]: Good call!

```suggestion
```

**Review by owencraston — APPROVED** [review]
Thank you!

**Review by tommasini — APPROVED** [review]

**Pull request merged**